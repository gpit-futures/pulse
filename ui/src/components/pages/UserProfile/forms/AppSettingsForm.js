import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { Field, reduxForm } from 'redux-form'
import { Row, Col } from 'react-bootstrap';

import ValidatedInput from '../../../form-fields/ValidatedInputFormGroup';
import FileInput from '../../../form-fields/FileInput';
import { optionsForThemesField } from './options-for-select.config';
import Select from '../../../form-fields/SelectFormGroup';
import { validateAppSettingsForm } from '../forms.validation';
import { valuesSettingsForm, valuesSettingsFormLabels } from '../forms.config';
import { setTheme } from '../../../../ducks/set-theme.duck';

@reduxForm({
  form: 'appSettingsFormSelector',
  validate: validateAppSettingsForm,
})
export default class AppSettingsForm extends PureComponent {
  static propTypes = {
    dispatch: PropTypes.func,
    initialize: PropTypes.func,
  };

  componentDidMount() {
    const { patientsInfo, initialize } = this.props;
    initialize(this.defaultValuesForm(patientsInfo));
  }

  defaultValuesForm(value) {
    const defaultFormValues = {
      [valuesSettingsForm.APP_TITLE]: value.title,
      [valuesSettingsForm.LOGO_PATH]: value.logoB64,
      [valuesSettingsForm.SELECT_THEME]: value.themeColor,
      [valuesSettingsForm.BROWSER_TITLE]: value.browserTitle,
    };

    return defaultFormValues;
  }

 setThemeHook = dispatch => (theme) => {
   dispatch(setTheme(theme));
   return theme;
 };

 render() {
   const { dispatch } = this.props;

   return (
     <div className="panel-body-inner">
       <form name="appSettingsForm" className="form">
         <div className="form-group-wrapper">
           <Row>
             <Col xs={12} md={6}>
               <Row>
                 <Col md={11}>
                   <Field
                     label={valuesSettingsFormLabels.APP_TITLE}
                     name={valuesSettingsForm.APP_TITLE}
                     type="text"
                     placeholder=""
                     component={ValidatedInput}
                   />
                   <Field
                     label={valuesSettingsFormLabels.LOGO_PATH}
                     name={valuesSettingsForm.LOGO_PATH}
                     component={FileInput}
                     id={valuesSettingsForm.LOGO_PATH}
                   />
                   <Field
                     label={valuesSettingsFormLabels.SELECT_THEME}
                     name={valuesSettingsForm.SELECT_THEME}
                     component={Select}
                     options={optionsForThemesField}
                     //here we're using normalize hook for dispatching action to set app theme
                     normalize={this.setThemeHook(dispatch)}
                   />
                   <Field
                     label={valuesSettingsFormLabels.BROWSER_TITLE}
                     name={valuesSettingsForm.BROWSER_TITLE}
                     type="text"
                     placeholder=""
                     component={ValidatedInput}
                   />
                 </Col>
               </Row>
             </Col>
           </Row>
         </div>
       </form>
     </div>)
 }
}
