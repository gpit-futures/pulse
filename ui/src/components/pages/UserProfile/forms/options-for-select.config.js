const optionsForThemesField = [{
  value: 'green',
  title: 'Green Theme',
},
{
  value: 'purple',
  title: 'Purple Theme',
},
{
  value: 'gray',
  title: 'Gray Theme',
}];

const optionsForGenderField = [{
  value: 'female',
  title: 'Female',
}, {
  value: 'male',
  title: 'Male',
}];

const optionsForCountryField = [{
  value: 'usa',
  title: 'USA',
}, {
  value: 'country1',
  title: 'Country 1',
}, {
  value: 'country2',
  title: 'Country 2',
}, {
  value: 'country3',
  title: 'Country 3',
}, {
  value: 'country4',
  title: 'Country 4',
}, {
  value: 'country5',
  title: 'Country 5',
}];

export { optionsForThemesField, optionsForGenderField, optionsForCountryField }
