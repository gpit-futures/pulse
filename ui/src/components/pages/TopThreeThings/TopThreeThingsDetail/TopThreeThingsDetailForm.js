import React, { PureComponent } from 'react';
import { Field, reduxForm } from 'redux-form'

import ValidatedInput from '../../../form-fields/ValidatedInputFormGroup';
import ValidatedTextareaFormGroup from '../../../form-fields/ValidatedTextareaFormGroup';
import DateInput from '../../../form-fields/DateInput';
import { valuesNames, valuesLabels } from '../forms.config';
import { validateForm } from '../forms.validation';
import { getDDMMMYYYY } from '../../../../utils/time-helpers.utils';
import FormSectionList from '../../../form-fields/FormSectionList';
import FormSection from '../../../form-fields/FormSection';

@reduxForm({
  form: 'topThreeThingsPanelFormSelector',
  validate: validateForm,
})
export default class TopThreeThingsDetailForm extends PureComponent {
  componentDidMount() {
    const { detail, initialize } = this.props;
    initialize(this.defaultValuesForm(detail));
  }
  defaultValuesForm(value) {
    const defaultFormValues = {
      [valuesNames.NAME1]: value[valuesNames.NAME1],
      [valuesNames.NAME2]: value[valuesNames.NAME2],
      [valuesNames.NAME3]: value[valuesNames.NAME3],
      [valuesNames.DESCRIPTION1]: value[valuesNames.DESCRIPTION1],
      [valuesNames.DESCRIPTION2]: value[valuesNames.DESCRIPTION2],
      [valuesNames.DESCRIPTION3]: value[valuesNames.DESCRIPTION3],
      [valuesNames.SOURCE]: value[valuesNames.SOURCE],
    };

    return defaultFormValues;
  }
  render() {
    const { isSubmit } = this.props;
    const date = new Date();
    const dateCreated = getDDMMMYYYY(date.getTime());
    return (
      <div className="panel-body-inner">
        <form name="topThreeThingsPanelForm" className="form">
          <div className="form-group-wrapper">
            <FormSectionList title="Isssue #1">
              <FormSection isBordered theme='primary'>
                <div className="form-group-wrapper">
                  <div className="row-expand">
                    <div className="col-expand-left">
                      <Field
                        label={valuesLabels.NAME}
                        name={valuesNames.NAME1}
                        id={valuesNames.NAME1}
                        type="text"
                        placeholder=""
                        component={ValidatedInput}
                        props={{ isSubmit }}
                      />
                    </div>
                    <div className="col-expand-right">
                      <Field
                        label={valuesLabels.DESCRIPTION}
                        name={valuesNames.DESCRIPTION1}
                        id={valuesNames.DESCRIPTION1}
                        component={ValidatedTextareaFormGroup}
                        props={{ isSubmit }}
                      />
                    </div>
                  </div>
                </div>
              </FormSection>
            </FormSectionList>
            <FormSectionList title="Isssue #2">
              <FormSection isBordered theme='primary'>
                <div className="form-group-wrapper">
                  <div className="row-expand">
                    <div className="col-expand-left">
                      <Field
                        label={valuesLabels.NAME}
                        name={valuesNames.NAME2}
                        id={valuesNames.NAME2}
                        type="text"
                        placeholder=""
                        component={ValidatedInput}
                        props={{ isSubmit }}
                      />
                    </div>
                    <div className="col-expand-right">
                      <Field
                        label={valuesLabels.DESCRIPTION}
                        name={valuesNames.DESCRIPTION2}
                        id={valuesNames.DESCRIPTION2}
                        component={ValidatedTextareaFormGroup}
                        props={{ isSubmit }}
                      />
                    </div>
                  </div>
                </div>
              </FormSection>
            </FormSectionList>
            <FormSectionList title="Isssue #3">
              <FormSection isBordered theme='primary'>
                <div className="form-group-wrapper">
                  <div className="row-expand">
                    <div className="col-expand-left">
                      <Field
                        label={valuesLabels.NAME}
                        name={valuesNames.NAME3}
                        id={valuesNames.NAME3}
                        type="text"
                        placeholder=""
                        component={ValidatedInput}
                        props={{ isSubmit }}
                      />
                    </div>
                    <div className="col-expand-right">
                      <Field
                        label={valuesLabels.DESCRIPTION}
                        name={valuesNames.DESCRIPTION3}
                        id={valuesNames.DESCRIPTION3}
                        component={ValidatedTextareaFormGroup}
                        props={{ isSubmit }}
                      />
                    </div>
                  </div>
                </div>
              </FormSection>
            </FormSectionList>
            <div className="row-expand">
              <div className="col-expand-left">
                <Field
                  label={valuesLabels.DATE}
                  name={valuesNames.DATE}
                  id={valuesNames.DATE}
                  component={DateInput}
                  props={{ disabled: true, value: dateCreated, format: 'DD-MMM-YYYY' }}
                />
              </div>
              <div className="col-expand-right">
                <Field
                  label={valuesLabels.SOURCE}
                  name={valuesNames.SOURCE}
                  id={valuesNames.SOURCE}
                  component={ValidatedInput}
                  props={{ disabled: true }}
                />
              </div>
            </div>
          </div>
        </form>
      </div>)
  }
}
