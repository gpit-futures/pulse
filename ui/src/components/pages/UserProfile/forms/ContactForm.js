import React, { PureComponent } from 'react';
import { Field, reduxForm } from 'redux-form'
import { Row, Col } from 'react-bootstrap';

import ValidatedInput from '../../../form-fields/ValidatedInputFormGroup';
import { optionsForCountryField } from './options-for-select.config';
import Select from '../../../form-fields/SelectFormGroup';
import { validateContactForm } from '../forms.validation';
import { valuesContactForm, valuesContactFormLabels } from '../forms.config';
import { defaultContactFormValues } from './default-values.config';

@reduxForm({
  form: 'contactFormSelector',
  validate: validateContactForm,
})
export default class ContactForm extends PureComponent {
  componentDidMount() {
    this.props.initialize(defaultContactFormValues);
  }

  render() {
    return (
      <div className="panel-body-inner">
        <form name="contactForm" className="form">
          <div className="form-group-wrapper">
            <Row>
              <Col xs={12} md={6}>
                <Row>
                  <Col md={11}>
                    <Field
                      label={valuesContactFormLabels.ADDRESS}
                      name={valuesContactForm.ADDRESS}
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                    <Field
                      label={valuesContactFormLabels.CITY}
                      name={valuesContactForm.CITY}
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                    <Field
                      label={valuesContactFormLabels.STATE}
                      name={valuesContactForm.STATE}
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                    <Field
                      label={valuesContactFormLabels.POSTAL_CODE}
                      name={valuesContactForm.POSTAL_CODE}
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                    <Field
                      label={valuesContactFormLabels.SELECT_COUNTRY}
                      name={valuesContactForm.SELECT_COUNTRY}
                      placeholder=""
                      component={Select}
                      options={optionsForCountryField}
                    />
                  </Col>
                </Row>
              </Col>
              <Col xs={12} md={6}>
                <Row>
                  <Col md={11} mdOffset={1}>
                    <Field
                      label={valuesContactFormLabels.PHONE}
                      name={valuesContactForm.PHONE}
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                    <Field
                      label={valuesContactFormLabels.EMAIL}
                      name={valuesContactForm.EMAIL}
                      placeholder=""
                      component={ValidatedInput}
                    />
                  </Col>
                </Row>
              </Col>
            </Row>
          </div>
        </form>
      </div>)
  }
}
