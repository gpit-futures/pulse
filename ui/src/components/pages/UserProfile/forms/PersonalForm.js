import React, { PureComponent } from 'react';
import { Field, reduxForm } from 'redux-form'
import { Row, Col } from 'react-bootstrap';

import ValidatedInput from '../../../form-fields/ValidatedInputFormGroup';
import DateInput from '../../../form-fields/DateInput';
import { optionsForGenderField } from './options-for-select.config';
import Select from '../../../form-fields/SelectFormGroup';
import { validatePersonalForm } from '../forms.validation';
import { valuesPersonalForm, valuesPersonalFormLabels } from '../forms.config';
import { defaultPersonalFormValues } from './default-values.config';

@reduxForm({
  form: 'personalFormSelector',
  validate: validatePersonalForm,
})
export default class PersonalForm extends PureComponent {
  componentDidMount() {
    this.props.initialize(defaultPersonalFormValues);
  }

  render() {
    return (
      <div className="panel-body-inner">
        <form name="personalForm" className="form">
          <div className="form-group-wrapper">
            <Row>
              <Col xs={12} md={6}>
                <Row>
                  <Col md={11}>
                    <Field
                      label={valuesPersonalFormLabels.FIRST_NAME}
                      name={valuesPersonalForm.FIRST_NAME}
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                    <Field
                      label={valuesPersonalFormLabels.LAST_NAME}
                      name={valuesPersonalForm.LAST_NAME}
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                    <Field
                      label={valuesPersonalFormLabels.NHS_NUMBER}
                      name={valuesPersonalForm.NHS_NUMBER}
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                  </Col>
                </Row>
              </Col>
              <Col xs={12} md={6}>
                <Row>
                  <Col md={11} mdOffset={1}>
                    <Field
                      label={valuesPersonalFormLabels.DATE_OF_BIRTH}
                      name={valuesPersonalForm.DATE_OF_BIRTH}
                      type="text"
                      placeholder=""
                      component={DateInput}
                      props={{ format: 'DD-MMM-YYYY' }}
                    />
                    <Field
                      label={valuesPersonalFormLabels.SELECT_GENDER}
                      name={valuesPersonalForm.SELECT_GENDER}
                      placeholder=""
                      component={Select}
                      options={optionsForGenderField}
                    />
                    <Field
                      label={valuesPersonalFormLabels.DOCTOR}
                      name={valuesPersonalForm.DOCTOR}
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                  </Col>
                </Row>
              </Col>
            </Row>
          </div>
        </form>
      </div>)
  }
}
