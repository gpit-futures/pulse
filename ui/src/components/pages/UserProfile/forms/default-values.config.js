import { valuesSettingsForm, valuesPersonalForm, valuesContactForm } from '../forms.config';

const defaultPersonalFormValues = {
  [valuesPersonalForm.FIRST_NAME]: 'Bob',
  [valuesPersonalForm.LAST_NAME]: 'Smith',
  [valuesPersonalForm.SELECT_GENDER]: 'female',
  [valuesPersonalForm.DOCTOR]: 'Dr Emma Huston',
  [valuesPersonalForm.DATE_OF_BIRTH]: 0,
};

const defaultContactFormValues = {
  [valuesContactForm.ADDRESS]: '6801 Tellus Street',
  [valuesContactForm.CITY]: 'Westmorland',
  [valuesContactForm.STATE]: 'Westmorland',
  [valuesContactForm.POSTAL_CODE]: 'Box 306',
  [valuesContactForm.PHONE]: '07624 647524',
  [valuesContactForm.EMAIL]: 'bob.smith@gmail.com',
};

const defaultAppSettingsFormValues = {
  [valuesSettingsForm.APP_TITLE]: 'should be fetched from server',
  [valuesSettingsForm.LOGO_PATH]: 'should be fetched from server',
  [valuesSettingsForm.SELECT_THEME]: 'themeDefault',
  [valuesSettingsForm.BROWSER_TITLE]: 'should be fetched from server',
};

export { defaultPersonalFormValues, defaultContactFormValues, defaultAppSettingsFormValues }
