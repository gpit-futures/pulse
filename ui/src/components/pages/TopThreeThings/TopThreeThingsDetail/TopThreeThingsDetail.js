import React, { PureComponent } from 'react';

import PluginDetailPanel from '../../../plugin-page-component/PluginDetailPanel'
import TopThreeThingsDetailForm from './TopThreeThingsDetailForm'
import { getDDMMMYYYY } from '../../../../utils/time-helpers.utils';
import { valuesNames, valuesLabels } from '../forms.config';
import FormSectionList from '../../../form-fields/FormSectionList';
import FormSection from '../../../form-fields/FormSection';

const TOP_THREE_THINGS_PANEL = 'topThreeThingsPanel';

export default class TopThreeThingsDetail extends PureComponent {
  render() {
    const { onExpand, openedPanel, expandedPanel, currentPanel, onEdit, editedPanel, onCancel, onSaveSettings, topThreeThingFormValues, isSubmit } = this.props;
    let { detail } = this.props;
    detail = detail || {};
    const dateCreated = getDDMMMYYYY(detail.dateCreated);

    return (
      <div className="section-detail">
        <div className="panel-group accordion">
          {(expandedPanel === TOP_THREE_THINGS_PANEL || expandedPanel === 'all') && !editedPanel[TOP_THREE_THINGS_PANEL] ? <PluginDetailPanel
            onExpand={onExpand}
            name={TOP_THREE_THINGS_PANEL}
            title="Top 3 Thing"
            isOpen={openedPanel === TOP_THREE_THINGS_PANEL}
            currentPanel={currentPanel}
            onEdit={onEdit}
            editedPanel={editedPanel}
            onCancel={onCancel}
            onSaveSettings={onSaveSettings}
            formValues={topThreeThingFormValues}
            isBtnShowPanel={false}
          >
            <div className="panel-body-inner">
              <div className="form">
                <div className="form-group-wrapper">
                  <div>
                    <FormSectionList title="Isssue #1">
                      <FormSection isBordered theme='primary'>
                        <div className="form-group-wrapper">
                          <div className="row-expand">
                            <div className="col-expand-left">
                              <div className="form-group">
                                <label className="control-label">{valuesLabels.NAME}</label>
                                <div className="form-control-static">{detail[valuesNames.NAME1]}</div>
                              </div>
                            </div>
                            <div className="col-expand-right">
                              <div className="form-group">
                                <label className="control-label">{valuesLabels.DESCRIPTION}</label>
                                <div className="form-control-static">{detail[valuesNames.DESCRIPTION1]}</div>
                              </div>
                            </div>
                          </div>
                        </div>
                      </FormSection>
                    </FormSectionList>
                    <FormSectionList title="Isssue #2">
                      <FormSection isBordered theme='primary'>
                        <div className="form-group-wrapper">
                          <div className="row-expand">
                            <div className="col-expand-left">
                              <div className="form-group">
                                <label className="control-label">{valuesLabels.NAME}</label>
                                <div className="form-control-static">{detail[valuesNames.NAME2]}</div>
                              </div>
                            </div>
                            <div className="col-expand-right">
                              <div className="form-group">
                                <label className="control-label">{valuesLabels.DESCRIPTION}</label>
                                <div className="form-control-static">{detail[valuesNames.DESCRIPTION2]}</div>
                              </div>
                            </div>
                          </div>
                        </div>
                      </FormSection>
                    </FormSectionList>
                    <FormSectionList title="Isssue #3">
                      <FormSection isBordered theme='primary'>
                        <div className="form-group-wrapper">
                          <div className="row-expand">
                            <div className="col-expand-left">
                              <div className="form-group">
                                <label className="control-label">{valuesLabels.NAME}</label>
                                <div className="form-control-static">{detail[valuesNames.NAME3]}</div>
                              </div>
                            </div>
                            <div className="col-expand-right">
                              <div className="form-group">
                                <label className="control-label">{valuesLabels.DESCRIPTION}</label>
                                <div className="form-control-static">{detail[valuesNames.DESCRIPTION3]}</div>
                              </div>
                            </div>
                          </div>
                        </div>
                      </FormSection>
                    </FormSectionList>
                    <div className="row-expand">
                      <div className="col-expand-left">
                        <div className="form-group">
                          <label className="control-label">{valuesLabels.DATE}</label>
                          <div className="form-control-static">{dateCreated}</div>
                        </div>
                      </div>
                      <div className="col-expand-right">
                        <div className="form-group">
                          <label className="control-label">{valuesLabels.SOURCE}</label>
                          <div className="form-control-static">{detail[valuesNames.SOURCE]}</div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </PluginDetailPanel> : null}
          {(expandedPanel === TOP_THREE_THINGS_PANEL || expandedPanel === 'all') && editedPanel[TOP_THREE_THINGS_PANEL] ? <PluginDetailPanel
            onExpand={onExpand}
            name={TOP_THREE_THINGS_PANEL}
            title="Edit Top 3 Thing"
            isOpen={openedPanel === TOP_THREE_THINGS_PANEL}
            currentPanel={currentPanel}
            onEdit={onEdit}
            editedPanel={editedPanel}
            onCancel={onCancel}
            onSaveSettings={onSaveSettings}
            formValues={topThreeThingFormValues}
            isBtnShowPanel={false}
          >
            <TopThreeThingsDetailForm
              detail={detail}
              isSubmit={isSubmit}
            />
          </PluginDetailPanel> : null}
        </div>
      </div>
    )
  }
}
