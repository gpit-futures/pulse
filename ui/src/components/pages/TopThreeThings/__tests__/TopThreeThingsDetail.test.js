import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';

import TopThreeThingsDetail from '../TopThreeThingsDetail/TopThreeThingsDetail';
import { valuesNames, valuesLabels } from '../forms.config';
import { getDDMMMYYYY } from '../../../../utils/time-helpers.utils';

Enzyme.configure({ adapter: new Adapter() });

const DATE_TO_USE = new Date('2018');
const DATE_TO_USE_TIME = DATE_TO_USE.getTime();
global.Date = jest.fn(() => DATE_TO_USE);

const TOP_THREE_THINGS_PANEL = 'topThreeThingsPanel';
const TOP_THREE_THINGS_DETAIL_TITLE = 'Top 3 Thing';
const TOP_THREE_THINGS_DETAIL_EDIT_TITLE = 'Edit Top 3 Thing';

const testProps = {
  onExpand: () => {},
  onEdit: () => {},
  onCancel: () => {},
  onSaveSettings: () => {},
  openedPanel: {},
  expandedPanel: 'all',
  currentPanel: 'currentPanel',
  editedPanel: 'editedPanel',
  topThreeThingFormValues: {},
};

const detail = {
  [valuesNames.SOURCE]: 'QEWDDB',
  [valuesNames.SOURCE_ID]: '26566e17-0ede-4818-8453-728ea0aa142c',
  [valuesNames.DATE_CREATED]: 1517475061744,
  [valuesNames.NAME1]: 'Item 2',
  [valuesNames.DESCRIPTION1]: 'My first problem',
  [valuesNames.NAME2]: 'Item 2',
  [valuesNames.DESCRIPTION2]: 'My second problem',
  [valuesNames.NAME3]: 'Item 2',
  [valuesNames.DESCRIPTION3]: '1',
};

const CONVERT_DATE_CREATED = getDDMMMYYYY(DATE_TO_USE_TIME);

describe('Component <TopThreeThingsDetail />', () => {
  it('should renders with all props correctly', () => {
    const component = shallow(
      <TopThreeThingsDetail
        detail={detail}
        onExpand={testProps.onExpand}
        onEdit={testProps.onEdit}
        onCancel={testProps.onCancel}
        onSaveSettings={testProps.onSaveSettings}
        openedPanel={testProps.openedPanel}
        expandedPanel={testProps.expandedPanel}
        currentPanel={testProps.currentPanel}
        editedPanel={testProps.editedPanel}
        topThreeThingFormValues={testProps.topThreeThingFormValues}
        isSubmit={false}
      />
    );

    expect(component).toMatchSnapshot();

    // Testing component when detail filled object, expandedPanel is all, and panel not edited
    expect(component.instance().props.detail).toEqual(detail);
    expect(component.instance().props.onExpand).toEqual(testProps.onExpand);
    expect(component.instance().props.onEdit).toEqual(testProps.onEdit);
    expect(component.instance().props.onCancel).toEqual(testProps.onCancel);
    expect(component.instance().props.onSaveSettings).toEqual(testProps.onSaveSettings);
    expect(component.instance().props.openedPanel).toEqual(testProps.openedPanel);
    expect(component.instance().props.expandedPanel).toEqual(testProps.expandedPanel);
    expect(component.instance().props.currentPanel).toEqual(testProps.currentPanel);
    expect(component.instance().props.editedPanel).toEqual(testProps.editedPanel);
    expect(component.instance().props.topThreeThingFormValues).toEqual(testProps.topThreeThingFormValues);
    expect(component.instance().props.isSubmit).toEqual(false);

    expect(component.find('.section-detail')).toHaveLength(1);
    expect(component.find('.form')).toHaveLength(1);
    expect(component.find('PluginDetailPanel')).toHaveLength(1);
    expect(component.find('FormSectionList')).toHaveLength(3);
    expect(component.find('FormSectionList')).toHaveLength(3);
    expect(component.find('FormSection')).toHaveLength(3);

    expect(component.find('PluginDetailPanel').props().currentPanel).toEqual(testProps.currentPanel);
    expect(component.find('PluginDetailPanel').props().editedPanel).toEqual(testProps.editedPanel);
    expect(component.find('PluginDetailPanel').props().isBtnShowPanel).toEqual(false);
    expect(component.find('PluginDetailPanel').props().isOpen).toEqual(testProps.editedPanel === TOP_THREE_THINGS_PANEL);
    expect(component.find('PluginDetailPanel').props().name).toEqual(TOP_THREE_THINGS_PANEL);
    expect(component.find('PluginDetailPanel').props().onCancel).toEqual(testProps.onCancel);
    expect(component.find('PluginDetailPanel').props().onEdit).toEqual(testProps.onEdit);
    expect(component.find('PluginDetailPanel').props().onExpand).toEqual(testProps.onExpand);
    expect(component.find('PluginDetailPanel').props().onSaveSettings).toEqual(testProps.onSaveSettings);
    expect(component.find('PluginDetailPanel').props().title).toEqual(TOP_THREE_THINGS_DETAIL_TITLE);


    expect(component.find('.control-label').at(0).text()).toEqual(valuesLabels.NAME);
    expect(component.find('.control-label').at(1).text()).toEqual(valuesLabels.DESCRIPTION);
    expect(component.find('.control-label').at(2).text()).toEqual(valuesLabels.NAME);
    expect(component.find('.control-label').at(3).text()).toEqual(valuesLabels.DESCRIPTION);
    expect(component.find('.control-label').at(4).text()).toEqual(valuesLabels.NAME);
    expect(component.find('.control-label').at(5).text()).toEqual(valuesLabels.DESCRIPTION);
    expect(component.find('.control-label').at(6).text()).toEqual(valuesLabels.DATE);
    expect(component.find('.control-label').at(7).text()).toEqual(valuesLabels.SOURCE);

    expect(component.find('.form-control-static').at(0).text()).toEqual(detail[valuesNames.NAME1]);
    expect(component.find('.form-control-static').at(1).text()).toEqual(detail[valuesNames.DESCRIPTION1]);
    expect(component.find('.form-control-static').at(2).text()).toEqual(detail[valuesNames.NAME2]);
    expect(component.find('.form-control-static').at(3).text()).toEqual(detail[valuesNames.DESCRIPTION2]);
    expect(component.find('.form-control-static').at(4).text()).toEqual(detail[valuesNames.NAME3]);
    expect(component.find('.form-control-static').at(5).text()).toEqual(detail[valuesNames.DESCRIPTION3]);
    expect(component.find('.form-control-static').at(6).text()).toEqual(CONVERT_DATE_CREATED);
    expect(component.find('.form-control-static').at(7).text()).toEqual(detail[valuesNames.SOURCE]);
  });

  it('should renders correctly with different state of props', () => {
    const component = shallow(
      <TopThreeThingsDetail
        expandedPanel={'all'}
        editedPanel={{ [TOP_THREE_THINGS_PANEL]: true }}
        openedPanel={{ [TOP_THREE_THINGS_PANEL]: true }}
      />
    );
    expect(component).toMatchSnapshot();

    // Testing component when detail filled object, expandedPanel is all, and panel is edited
    component.setProps({ detail });
    expect(component).toMatchSnapshot();

    expect(component.find('PluginDetailPanel')).toHaveLength(1);
    expect(component.find('ReduxForm')).toHaveLength(1);
    expect(component.find('PluginDetailPanel').props().title).toEqual(TOP_THREE_THINGS_DETAIL_EDIT_TITLE);
  });
});

