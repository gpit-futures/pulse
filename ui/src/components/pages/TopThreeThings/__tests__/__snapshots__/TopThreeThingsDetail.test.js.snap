// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component <TopThreeThingsDetail /> should renders correctly with different state of props 1`] = `
<div
  className="section-detail"
>
  <div
    className="panel-group accordion"
  >
    <PluginDetailPanel
      editedPanel={
        Object {
          "topThreeThingsPanel": true,
        }
      }
      isBtnShowPanel={false}
      isOpen={false}
      isShowControlPanel={true}
      name="topThreeThingsPanel"
      title="Edit Top 3 Thing"
    >
      <ReduxForm
        detail={Object {}}
      />
    </PluginDetailPanel>
  </div>
</div>
`;

exports[`Component <TopThreeThingsDetail /> should renders correctly with different state of props 2`] = `
<div
  className="section-detail"
>
  <div
    className="panel-group accordion"
  >
    <PluginDetailPanel
      editedPanel={
        Object {
          "topThreeThingsPanel": true,
        }
      }
      isBtnShowPanel={false}
      isOpen={false}
      isShowControlPanel={true}
      name="topThreeThingsPanel"
      title="Edit Top 3 Thing"
    >
      <ReduxForm
        detail={
          Object {
            "description1": "My first problem",
            "description2": "My second problem",
            "description3": "1",
            "name1": "Item 2",
            "name2": "Item 2",
            "name3": "Item 2",
            "source": "QEWDDB",
            "sourceId": "26566e17-0ede-4818-8453-728ea0aa142c",
            "undefined": 1517475061744,
          }
        }
      />
    </PluginDetailPanel>
  </div>
</div>
`;

exports[`Component <TopThreeThingsDetail /> should renders with all props correctly 1`] = `
<div
  className="section-detail"
>
  <div
    className="panel-group accordion"
  >
    <PluginDetailPanel
      currentPanel="currentPanel"
      editedPanel="editedPanel"
      formValues={Object {}}
      isBtnShowPanel={false}
      isOpen={false}
      isShowControlPanel={true}
      name="topThreeThingsPanel"
      onCancel={[Function]}
      onEdit={[Function]}
      onExpand={[Function]}
      onSaveSettings={[Function]}
      title="Top 3 Thing"
    >
      <div
        className="panel-body-inner"
      >
        <div
          className="form"
        >
          <div
            className="form-group-wrapper"
          >
            <div>
              <FormSectionList
                title="Isssue #1"
              >
                <FormSection
                  isAccordion={false}
                  isBordered={true}
                  isImportBtn={false}
                  theme="primary"
                >
                  <div
                    className="form-group-wrapper"
                  >
                    <div
                      className="row-expand"
                    >
                      <div
                        className="col-expand-left"
                      >
                        <div
                          className="form-group"
                        >
                          <label
                            className="control-label"
                          >
                            Name
                          </label>
                          <div
                            className="form-control-static"
                          >
                            Item 2
                          </div>
                        </div>
                      </div>
                      <div
                        className="col-expand-right"
                      >
                        <div
                          className="form-group"
                        >
                          <label
                            className="control-label"
                          >
                            Description
                          </label>
                          <div
                            className="form-control-static"
                          >
                            My first problem
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </FormSection>
              </FormSectionList>
              <FormSectionList
                title="Isssue #2"
              >
                <FormSection
                  isAccordion={false}
                  isBordered={true}
                  isImportBtn={false}
                  theme="primary"
                >
                  <div
                    className="form-group-wrapper"
                  >
                    <div
                      className="row-expand"
                    >
                      <div
                        className="col-expand-left"
                      >
                        <div
                          className="form-group"
                        >
                          <label
                            className="control-label"
                          >
                            Name
                          </label>
                          <div
                            className="form-control-static"
                          >
                            Item 2
                          </div>
                        </div>
                      </div>
                      <div
                        className="col-expand-right"
                      >
                        <div
                          className="form-group"
                        >
                          <label
                            className="control-label"
                          >
                            Description
                          </label>
                          <div
                            className="form-control-static"
                          >
                            My second problem
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </FormSection>
              </FormSectionList>
              <FormSectionList
                title="Isssue #3"
              >
                <FormSection
                  isAccordion={false}
                  isBordered={true}
                  isImportBtn={false}
                  theme="primary"
                >
                  <div
                    className="form-group-wrapper"
                  >
                    <div
                      className="row-expand"
                    >
                      <div
                        className="col-expand-left"
                      >
                        <div
                          className="form-group"
                        >
                          <label
                            className="control-label"
                          >
                            Name
                          </label>
                          <div
                            className="form-control-static"
                          >
                            Item 2
                          </div>
                        </div>
                      </div>
                      <div
                        className="col-expand-right"
                      >
                        <div
                          className="form-group"
                        >
                          <label
                            className="control-label"
                          >
                            Description
                          </label>
                          <div
                            className="form-control-static"
                          >
                            1
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </FormSection>
              </FormSectionList>
              <div
                className="row-expand"
              >
                <div
                  className="col-expand-left"
                >
                  <div
                    className="form-group"
                  >
                    <label
                      className="control-label"
                    >
                      Date
                    </label>
                    <div
                      className="form-control-static"
                    >
                      01-Jan-2018
                    </div>
                  </div>
                </div>
                <div
                  className="col-expand-right"
                >
                  <div
                    className="form-group"
                  >
                    <label
                      className="control-label"
                    >
                      Source
                    </label>
                    <div
                      className="form-control-static"
                    >
                      QEWDDB
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </PluginDetailPanel>
  </div>
</div>
`;
